### PR TITLE
Add checking datasets before each tailored download

### LINKED ISSUE #189: Speed up 15 mins
## Detailed Description
I'm worried duplicated customization are being made, and that really slows the process of getting all the 15 mins data. 

Currently at the start of the script we check to see if the files have already been made. 
I wonder before each customization is made, could the app check to see if the file is in s3, if it is, load it, if not run the customization?

## Context
- when the satellite is on backup, it takes about 30 mins to get 1 hour of data. 

## Possible Implementation
See above

### PR DESCRIPTION
**Pull request by jacobbieker**
Also small refactor to make name of function more pythonic

# Pull Request

## Description

Checks each dataset when using the Data Tailor to see if its already been downloaded, before starting. This should hopefully reduce how long it takes to download the 15 minute data, and reduce duplication of work.

Fixes #189 

## How Has This Been Tested?

Unit tests

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: satip/app.py [hunk 1]
```diff
@@ -170,26 +171,30 @@ def run(
...
                 random.shuffle(datasets) # Shuffle so subsequent runs might download different data
                 updated_data = True
                 if use_backup:
-                    download_manager.download_tailored_datasets(
-                        datasets,
-                        product_id="EO:EUM:DAT:MSG:HRSEVIRI",
-                    )
+                    # Check before downloading each tailored dataset, as it can take awhile
+                    for dset in datasets:
+                        dset = utils.filter_dataset_ids_on_current_files([dset], save_dir)
```
Comment (peterdudfield) [inline satip/app.py:192]: do the current files get re loaded when this is called?

Comment (peterdudfield) [reply satip/app.py:192]: I think it does, this looks good

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
I've approved, the tests are failing but this is due to 5 min satellite down at the moment, i think

**Comment by peterdudfield** [comment]
Should we do this at the start of getting the 5 mins data aswell? So if another Consumer starts running (dataset order is randomized) but then once we start putting the next image, we check if its downlaoded by the other consumer in the meantime?

**Comment by jacobbieker** [comment]
Yeah, that's probably worth it

**Comment by jacobbieker** [comment]
@peterdudfield tests now pass and it does it for both 5 minutely and 15 minutely data

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**